### PR TITLE
Add agent cert reissue (CSR) back in as step 4

### PR DESCRIPTION
**Pull request by jeffmccune**
This patch adds the step to reissue a new SSL certificate to the agent
system.  This step was previous removed when refactoring out steps 1 and
2 to focus on securing the fleet only.

Step 4 will help customers clean up agent certificates which may
incorrectly have subject alternative names in them.  The clean up
process involves issuing new certificates from a brand new CA created
and configured in step 3.

This patch also introduced the behavior where we no longer turn off
certificate revocation checks on the agent.  We don't need to do this
anymore once this patch is applied because the catalog that triggers the
new CSR _also_ gives the agent the current CRL and CA certificate of the
new CA.  This means the agent will "just work" once the master is using
a certificate issued by the new CA.

However, this also means the node will refuse to talk to the master
until the master completes step 5.  In step 5 the master will replace
the SSL certificate issued by the previous CA with a SSL certificate
issued by the new CA.  The agents will no longer trust the previous CA
after step 4, causing them to be "disconnected."

### CODE REVIEW

File: bin/pe_step4_migrate_agents_to_new_authority [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export PATH="/opt/puppet/bin:$PATH"
+module="cve20113872"
+class="${module}::step4"
+
+manifest="$(puppet master --configprint manifest)"
+
+timestamp="$(ruby -e 'puts Time.now.to_i')"
+idx=0
+
+# Add the cve20113872 class to every nodes catalog.  This class will actually flip the node
```
Comment (nigelkersten) [inline bin/pe_step4_migrate_agents_to_new_authority]: Correction: Add the cve20113872::step4 class.

File: bin/pe_step4_migrate_agents_to_new_authority [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+fi
+
+cat <<EOMESSAGE
+
+The Puppet Master has been reconfigured to include class ${class} in all node
+catalogs.
+
+This class will configure each agent system to generate a new CSR and submit it
+to the master.  The new certificate needs to be signed for the agent to operate
+again.
```
Comment (nigelkersten) [inline bin/pe_step4_migrate_agents_to_new_authority:41]: We should also mention here that we're orphaning the nodes until step5 is completed on the master.

File: bin/pe_step4_migrate_agents_to_new_authority [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+class="${module}::step4"
+
+manifest="$(puppet master --configprint manifest)"
+
+timestamp="$(ruby -e 'puts Time.now.to_i')"
+idx=0
+
+# Add the cve20113872 class to every nodes catalog.  This class will actually flip the node
+# to the new DNS name and not do anything with the client certificate.
+# FIXME: The class needs to be updated to only flip the server setting in puppet.conf
```
Comment (nigelkersten) [inline bin/pe_step4_migrate_agents_to_new_authority]: Is that FIXME still current?

File: bin/pe_step4_migrate_agents_to_new_authority (hunk 2 above)
Comment (jeffmccune) [inline bin/pe_step4_migrate_agents_to_new_authority:41]: I addressed this comment by amending the commit.

File: bin/pe_step4_migrate_agents_to_new_authority (hunk 1 above)
Comment (jeffmccune) [inline bin/pe_step4_migrate_agents_to_new_authority]: Corrected by amending the commit.

File: bin/pe_step4_migrate_agents_to_new_authority (hunk 3 above)
Comment (jeffmccune) [inline bin/pe_step4_migrate_agents_to_new_authority]: Removed the FIXME by amending the commit.  Also updated this comment to be accurate as of now.

**Pull request merged**